### ISSUE TITLE
IOS release size is too large after upgrading Flutter to 1.12.13+hotfix.7

### ISSUE THREAD

**Issue by vaidehee** [flutter/flutter#49855]
My iOS release size is unexpectedly too large after upgrading Flutter to 1.12.13+hotfix.7. And Itanagar more than 185+ Mb, this never happened before. I was used to work on lower version and it was about around 40 Mb. Can anybody help me out?  @filiph

**Comment by filiph** [comment]
Thanks @vaidehee for the report! Do you remember what the previous version (before upgrading to `1.12.13+hotfix.7`) was? It might also help if you run `flutter doctor -v` and paste the output here.

@cbracken Are we seeing a significant increase (4x) in iOS binary sizes with the latest Flutter stable?

**Comment by VladyslavBondarenko** [comment]
May be related to https://github.com/flutter/flutter/issues/47101

**Comment by maciz84** [comment]
I have the same issue? any news about this

**Comment by jonahwilliams** [comment]
See https://github.com/flutter/flutter/issues/47101#issuecomment-567522077

TLDR: Your app is not actually that big

**Comment by vaidehee** [comment]
sure @filiph ,
i was used to work on Flutter 1.9.1+hotfix.6  before upgrading ,
and this is is an output of flutter doctor.




[✓] Flutter (Channel stable, v1.12.13+hotfix.7, on Mac OS X 10.14.6 18G103, locale en-IN)
    • Flutter version 1.12.13+hotfix.7 at /Users/tristate/Flutter/flutter
    • Framework revision 9f5ff2306b (5 days ago), 2020-01-26 22:38:26 -0800
    • Engine revision a67792536c
    • Dart version 2.7.0

 
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/tristate/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 10.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2, Build version 10E125
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[!] VS Code (version 1.37.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Redmi 4                   • f725aa8b7ce4                         • android-arm64 • Android 7.1.2 (API 25)
    • Android SDK built for x86 • emulator-5554                        • android-x86   • Android 9 (API 28) (emulator)
    • iPhone 8                  • 38B12C2F-B971-4943-AED6-90EC4BA57F22 • ios           • com.apple.CoreSimulator.SimRuntime.iOS-12-2 (simulator)

**Comment by vaidehee** [comment]
> See [#47101 (comment)](https://github.com/flutter/flutter/issues/47101#issuecomment-567522077)
> 
> TLDR: Your app is not actually that big

 yes, but as suggested in the comment i created distribution app and that too was too large,
this did never happen and if it is like so than even a simple "Hello world" app would have around 40+ mbs that is not at all good.

**Comment by filiph** [comment]
Just to be clear: you tried uploading the file ("Archive") to App Store Connect and it was declined for being too large?

As explained in the other thread, the increased file size is a natural consequence of having bit code in the intermediate file that you are sending to Apple. It is not indicative of what the user will download.

**Comment by vaidehee** [comment]
> Just to be clear: you tried uploading the file ("Archive") to App Store Connect and it was declined for being too large?
> 
> As explained in the other thread, the increased file size is a natural consequence of having bit code in the intermediate file that you are sending to Apple. It is not indicative of what the user will download.

I understand, but during the development of an application, we need to provide a build to client multiple times and he is rejecting this much of size of an IPA and questioning us why it was not happening before. 
may I know if this will be resolved in the upcoming version so we can explain it to them?

**Comment by filiph** [comment]
Oh, I understand. Thanks for the additional context.

This is not going away, and it's not specific to Flutter. Using LLVM IR will get you these sizes whether or not you use Flutter. As explained in the [linked FAQ](https://flutter.dev/docs/resources/faq#how-big-is-the-flutter-engine):

> As of Flutter SDK 1.12, the release engine binary now includes LLVM IR (bitcode). Xcode uses this bitcode to produce a final binary for the App Store containing the latest compiler optimizations and features.

For sending stuff to client, you might want to opt into sending profile builds. You might also want to experiment with https://flutter.dev/docs/perf/app-size#ios.

I'll close this issue. Feel free to reopen if you think this is in error.

**Comment by libleo** [comment]
> https://flutter.dev/docs/perf/app-size#ios

if I don't release my app to App Store, how to reduce bitcode part?

**Comment by vaidehee** [comment]
> > https://flutter.dev/docs/perf/app-size#ios
> 
> if I don't release my app to App Store, how to reduce bitcode part?

Hii, as Filiph suggested, I created the profile app using 
flutter build ios —profile,
And it works fine,
So during Development, use this command to build an ipa.

**Comment by yuhaocan** [comment]
> created the profile app using

how to created the profile app using
